src/com/thunder_cut/netio/DataReceiver.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * DataReceiver.java
+ * Author : Hyeokwoo Kwon
+ * Created Date : 2020-02-01
+ */
+package com.thunder_cut.netio;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+import java.util.EnumMap;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+/**
+ * Receives data sent from the server, looping until the program ends
+ */
+public class DataReceiver implements Runnable {
+
+    private enum HeaderItem {
+        TYPE,
+        SRC_ID,
+        DST_ID,
+        DATA_SIZE;
+    }
+
+    private SocketChannel socketChannel;
+
+    private BiConsumer<Integer, byte[]> drawImage;
+
+    private Map<Character, BiConsumer<Integer, byte[]>> dataType;
+
+    @Override
+    public void run() {
+
+        dataType = new HashMap<>();
+
+        dataType.put(DataType.IMG.type, drawImage);
+
+        while (true) {
+            DecapsulatedData decapsulatedData = receiveData();
+
+            dataType.get(decapsulatedData.dataType).accept(decapsulatedData.srcID, decapsulatedData.data.array());
```

Comment:
This will cause an bug.    
Use thread-safe data structure.    
or remove data structure, and use if expression.